models/demos/bert_tiny/tests/test_performance.py

```diff
@@ -0,0 +1,136 @@
+@pytest.mark.parametrize("batch_size", [8])
+@pytest.mark.parametrize("sequence_size", [128])
+@pytest.mark.parametrize("model_name", ["mrm8488/bert-tiny-finetuned-squadv2"])
+def test_perf_bert_tiny(
+    device,
+    batch_size,
+    sequence_size,
+    model_name,
+    model_location_generator,
+    reset_seeds,
+):
+    disable_persistent_kernel_cache()
+    model_name = str(model_location_generator(model_name, model_subdir="Bert"))
+    hugging_face_reference_model = BertForQuestionAnswering.from_pretrained(model_name, torchscript=False)
+    tokenizer_name = str(model_location_generator(model_name, model_subdir="Bert"))
+    config = hugging_face_reference_model.config
+    pytorch_model = hugging_face_reference_model
+
+    torch_bert_input = torch.randint(0, 100, (batch_size, sequence_size)).to(torch.int32)
+    torch_token_type_ids = torch.zeros((batch_size, sequence_size), dtype=torch.int32)
+    torch_position_ids = torch.zeros((batch_size, sequence_size), dtype=torch.int32)
+    torch_attention_mask = torch.zeros(1, sequence_size)
+
+    parameters = preprocess_model_parameters(
+        initialize_model=lambda: pytorch_model,
+        device=device,
+        convert_to_ttnn=lambda *_: True,
+    )
+
+    ttnn_bert_inputs = ttnn.from_torch(torch_bert_input, dtype=ttnn.uint32, device=device)
+    ttnn_token_type_ids = ttnn.from_torch(torch_token_type_ids, dtype=ttnn.uint32, device=device)
+    ttnn_position_ids = ttnn.from_torch(torch_position_ids, dtype=ttnn.uint32, device=device)
+    ttnn_attention_mask = ttnn.from_torch(torch_attention_mask, dtype=ttnn.bfloat16, device=device)
+
+    durations = []
+    for i in range(2):
+        start = time.time()
+        ttnn_output = bert_for_question_answering(
+            config,
+            input_ids=ttnn_bert_inputs,
+            token_type_ids=ttnn_token_type_ids,
+            position_ids=ttnn_position_ids,
+            attention_mask=ttnn_attention_mask,
+            parameters=parameters,
+            device=device,
+        )
+        output = ttnn.from_device(ttnn_output)
+
+        end = time.time()
+        durations.append(end - start)
+        enable_persistent_kernel_cache()
+
+    inference_and_compile_time, inference_time, *_ = durations
+
+    expected_compile_time, expected_inference_time = get_expected_times("bert_tiny")
+    prep_perf_report(
+        model_name="bert_tiny",
+        batch_size=batch_size,
+        inference_and_compile_time=inference_and_compile_time,
+        inference_time=inference_time,
+        expected_compile_time=expected_compile_time,
+        expected_inference_time=expected_inference_time,
+        comments="",
+        inference_time_cpu=0.0,
+    )
+
+    logger.info(f"Compile time: {inference_and_compile_time - inference_time}")
+    logger.info(f"Inference time: {inference_time}")
+    logger.info(f"Samples per second: {1 / inference_time * batch_size}")
```

Comment:
tests should fail if performance is lower than expected

Reply:
Updated the file with assert statements.